### PR TITLE
[luasocket] only works on windows

### PR DESCRIPTION
**Pull request by autoantwort**

### CODE REVIEW

File: ports/luasocket/portfile.cmake [hunk 1]
```diff
@@ -7,12 +7,11 @@ vcpkg_from_github(
 
 file(COPY ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt DESTINATION ${SOURCE_PATH})
 
-vcpkg_configure_cmake(
+vcpkg_cmake_configure(
     SOURCE_PATH ${SOURCE_PATH}
```
Comment (JonLiu1993) [inline ports/luasocket/portfile.cmake:11]:
```suggestion
    SOURCE_PATH "${SOURCE_PATH}"
```

Comment (JonLiu1993) [reply ports/luasocket/portfile.cmake:11]: @autoantwort ,Thanks for your pr,could you add double quotes around the path in the portfile.cmake file ?

File: scripts/ci.baseline.txt [hunk 2]
```diff
@@ -781,8 +781,6 @@ luafilesystem:arm-uwp=fail
 luafilesystem:x64-uwp=fail
 luasec:x64-windows-static=fail
 luasocket:arm-uwp=fail
-luasocket:x64-linux=fail
-luasocket:x64-osx=fail
 luasocket:x64-uwp=fail
```
Comment (JonLiu1993) [inline scripts/ci.baseline.txt:784]: @autoantwort ,If arm-uwp, x64-uwp triplet upstream are not supported, please delete it and use the supports field in vcpkg.json to indicate

**Review by JonLiu1993 — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
Thanks!

**Pull request merged**